modules/library/main/src/main/java/org/geotools/feature/simple/SimpleFeatureTypeBuilder.java

```diff
@@ -1050,10 +1065,80 @@ public static SimpleFeatureType retype(SimpleFeatureType original, Query query)
         return retype(original, query.getPropertyNames(), crs);
     }
 
+    /**
+     * Create a SimpleFeatureType with the parameters and coordinate reference system of the query.
+     *
+     * @param original SimpleFeatureType
+     * @param attributes name of attributes to include in result
+     * @param crs CoordinateReferenceSystem of result
+     * @return SimpleFeatureType updated with the provided CoordinateReferenceSystem
+     */
     private static SimpleFeatureType retype(
-            SimpleFeatureType original, String[] propertyNames, CoordinateReferenceSystem crs) {
-        // TODO Auto-generated method stub
-        return null;
+            SimpleFeatureType original, String[] attributes, CoordinateReferenceSystem crs) {
+        SimpleFeatureTypeBuilder b = new SimpleFeatureTypeBuilder();
+
+        // initialize the builder
+        b.init(original);
+
+        List<String> properties;
```

Comment:
In simple features we call these attributes. Actually, these are attribute names. But they are also provided as arguments, which makes things even more confusing. Maybe the argument should be called attributNames, and you can have the method just use arrays rather than a list? I promise, it's a bit old fashioned, but does not hurt :rofl:, and you can do for loops on them too.

Reply:
yeah you are right, fixed